libpod/networking_linux.go

```diff
@@ -222,29 +222,62 @@ func (r *RootlessNetNS) Do(toRun func() error) error {
 		// the link target will be available in the mount ns.
 		// see: https://github.com/containers/podman/issues/10855
 		resolvePath := "/etc/resolv.conf"
-		for i := 0; i < 255; i++ {
+		linkCount := 0
+		for i := 1; i < len(resolvePath); i++ {
 			// Do not use filepath.EvalSymlinks, we only want the first symlink under /run.
 			// If /etc/resolv.conf has more than one symlink under /run, e.g.
 			// -> /run/systemd/resolve/stub-resolv.conf -> /run/systemd/resolve/resolv.conf
 			// we would put the netns resolv.conf file to the last path. However this will
 			// break dns because the second link does not exists in the mount ns.
 			// see https://github.com/containers/podman/issues/11222
-			link, err := os.Readlink(resolvePath)
+			//
+			// We also need to resolve all path components not just the last file.
+			// see https://github.com/containers/podman/issues/12461
+
+			if resolvePath[i] != '/' {
```

Comment:
Can't we strip any potential trailing / and then do a `filepath.Dir()`?

Reply:
That would be the wrong order. You have to resolve from left to right.

Reply:
Maybe split the string on `/` to get an array of path components, and then iterate through them, adding each to the path and checking for a symlink?

Reply:
I do not see how this would be better than what I am doing now. This would just complicate things because I now have to constantly use strings.Split() and strings.Join().

Reply:
Fair - I just see the existing code as not exactly clear in what it's trying to do.

Reply:
If I have more time after podman 4.0 I will spend some time to add a test for this and maybe rework this to make it more readable.

In theory we could drop this when we are full netavark without iptables backend. In this case we would not need to bind mount /run and /var/lib/cni because netavark does not need write access to those directories.